utils/testHelper.js

```diff
@@ -2,53 +2,92 @@ const path = require('path')
 const normalizeNewline = require('normalize-newline')
 const fs = require('fs')
 const { spawn } = require('child_process')
-const assert = require('assert')
-const { TEST_STATUS } = require('./constants')
+
+const { TEST_STATUS, PATHS } = require('./constants')
+const parseList = require('./parseList')
+const log = require('./log')
+const settings = require('./settings')
 
 const readSampleData = dir => {
-	const sampleData = {}
-	fs.readdirSync(dir).forEach(filename => {
-		const { name, ext } = path.parse(filename)
-		const filePath = path.resolve(dir, filename)
+    const sampleData = {}
+    fs.readdirSync(dir).forEach(filename => {
+        const { name, ext } = path.parse(filename)
+        const filePath = path.resolve(dir, filename)
+
+        const data = fs.readFileSync(filePath).toString()
+        if (!sampleData[name]) sampleData[name] = {}
+
+        sampleData[name][ext] = data
+    })
+
+    return sampleData
+}
 
-		const data = fs.readFileSync(filePath).toString()
-		if (!sampleData[name]) sampleData[name] = {}
+const runTests = async (command, args, sampleData) => {
+    let count = 0
+    const results = new Array(Object.keys(sampleData).length)
+    for (let index = 0; index < results.length; index++) results[index] = {}
 
-		sampleData[name][ext] = data
-	})
+    log.notice('Running tests...')
+    for (const key in sampleData) {
+        const result = results[count]
+        const testCase = sampleData[key]['.in']
+        const expectedOutput = sampleData[key]['.ans']
 
-	return sampleData
+        const child = spawn(command, args)
+        child.stdin.end(testCase)
+
+        let output = ''
+        for await (const data of child.stdout) output += data.toString()
+
+        result['output'] = normalizeNewline(output)
+        result['expectedOutput'] = normalizeNewline(expectedOutput)
+
+        if (Object.is(result['output'], result['expectedOutput']))
+            result['status'] = TEST_STATUS.ACCEPTED
+        else
+            result['status'] = TEST_STATUS.WRONG_ANSWER
+        count++;
+    }
+    let prettyOutput = '\x1B[31m- actual \x1B[32m+ expected\x1B[0m\n'
+    results.forEach(r => {
```

Comment:
could just `join` these instead of `forEach`ing to a non-const

Reply:
Will so!